infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
> I suppose we shouldn't be discussing this without having that program text to look at.

Right, it would be much more convenient to discuss it having the concrete code. So if it can wait, I am totally for it. Ping me when it's ready.